src/python/foglamp/core/server.py

```diff
@@ -40,15 +52,73 @@ def _make_app():
         routes.setup(app)
         return app
 
+    @staticmethod
+    def _make_core_app():
+        """Creates the Service management REST server Core a.k.s service registry
+
+        :rtype: web.Application
+        """
+        app = web.Application(middlewares=[middleware.error_middleware])
+        routes.core_setup(app)
+        return app
+
     @classmethod
-    async def _start_scheduler(cls):
+    async def _start_scheduler(cls, core_mgt_port):
         """Starts the scheduler"""
-        cls.scheduler = Scheduler()
+        print("called _start_scheduler")
+        _logger.info("start scheduler")
+        cls.scheduler = Scheduler(core_mgt_port)
         await cls.scheduler.start()
 
     @classmethod
+    async def start_storage(cls, loop, host, m_port):
+        loop.call_soon(cls._start_storage(host, m_port))
+
+    @staticmethod
+    def _start_storage(host, m_port):
+        print("called _start_storage")
+        _logger.info("start storage")
+        try:
+            cmd_with_args = ['./storage', '--address={}'.format(host),
+                             '--port={}'.format(m_port)]
+            subprocess.call(cmd_with_args, cwd=_STORAGE_DIR)
+        except Exception as ex:
+            _logger.exception(str(ex))
+
+    @classmethod
+    def _start_core(cls, host,  service_port, management_port=0):
+        print("called _start_core")
+        _logger.info("start core")
+
+        ma = MultiApp()
+        ma.configure_app(cls._make_core_app(), host=host, port=management_port)
+        ma.configure_app(cls._make_app(), host=host, port=service_port)
+        # TODO: allow config / env var to set protocol
+        cls._register_core(host, management_port, service_port)
+        ma.run_all()
+
+    @classmethod
+    def _register_core(cls, host, mgt_port, service_port):
+        core_service_id = Service.Instances.register(name="FogLAMP Core", s_type="Core", address=host,
+                                                     port=service_port, management_port=mgt_port)
+
+        return core_service_id
+
+    # TODO: remove me | NOT NEEDED (hopefully, we shall be able to get info back from aiohttp)
```

Comment:
Yes, the port can be determined after the fact. See previous code posted in slack that does this.

Reply:
loop.run_forever()